### PR TITLE
Add IDEFICS Zero3 fine tuning script

### PR DESCRIPTION
**Pull request by HugoLaurencon**
# What does this PR do?

This PR adds an example to fine-tune IDEFICS using DeepSpeed Zero3.

## Who can review?

@LysandreJik

### CODE REVIEW

File: examples/idefics/idefics_zero3_finetuning.py [hunk 1]
```diff
@@ -0,0 +1,135 @@
+"""
+Launch with `deepspeed --num_gpus N idefics_zero3_finetuning.py`
+by replacing N by the number of your gpus
```
Comment (HugoLaurencon) [inline examples/idefics/idefics_zero3_finetuning.py:3]:
```suggestion
by replacing N with the number of your GPUs
```

File: examples/idefics/idefics_zero3_finetuning.py [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+    save_strategy="steps",
+    save_steps=40,
+    eval_steps=20,
+    logging_steps=20,
+    max_steps=40,
+    remove_unused_columns=False,
+    push_to_hub=False,
+    label_names=["labels"],
+    load_best_model_at_end=True,
+    report_to=None,
```
Comment (VictorSanh) [inline examples/idefics/idefics_zero3_finetuning.py:120]:
```suggestion
    report_to="none",
```

File: examples/idefics/idefics_zero3_finetuning.py [hunk 3]
```diff
@@ -0,0 +1,135 @@
+"""
+Launch with `deepspeed --num_gpus N idefics_zero3_finetuning.py`
+by replacing N with the number of your GPUs
```
Comment (VictorSanh) [inline examples/idefics/idefics_zero3_finetuning.py:3]: could you add the slurm script actually? it's not clear how to launch in multinode for the 80b right now
similar to https://gist.github.com/VictorSanh/710930406c8403a3bd3831e7ffba9afe

Comment (VictorSanh) [reply examples/idefics/idefics_zero3_finetuning.py:3]: also how about adding the requirements in that docstring too? the nice thing about notebooks is that you can create specific cells to install dependencies, but here it's a script, so either we have a `requirements.txt` file but it's kind of not the philosophy of that repo, or we write it somewhere as plain text

**Review by VictorSanh — COMMENTED** [review]
thank you Hugo! a few minor comments

File: examples/idefics/idefics_zero3_finetuning.py (hunk 3 above)
Comment (HugoLaurencon) [reply examples/idefics/idefics_zero3_finetuning.py]: Ok I made a folder instead with everything inside, let me know if it's more clear

**Pull request merged**